dali/operators/decoder/audio/generic_decoder.h

```diff
@@ -35,13 +36,16 @@ class DLL_PUBLIC GenericAudioDecoder : public TypedAudioDecoderBase<SampleType>
  public:
   DLL_PUBLIC GenericAudioDecoder();
 
+  DLL_PUBLIC GenericAudioDecoder(GenericAudioDecoder&&);
+  DLL_PUBLIC GenericAudioDecoder& operator=(GenericAudioDecoder&&);
+
```

Comment:
Per [rule of 6](https://en.cppreference.com/w/cpp/language/rule_of_three), you should define copy-ctors as well

Reply:
It's actually a _move_, not a copy - and I think it will work just fine by deleting the copy.

Reply:
I don't want copy constructor. I want them deleted

Reply:
> It's actually a move

Yeah, this one is move ;) But AFAIK, if you need any of custom ctor, dtor, copy, move, copy-assignment, move-assignment, good practice is to explicitly declare default, delete or implement them

> I don't want copy constructor. I want them deleted

That's fine. By "define" above I meant, that you should do something with them (deleting works just fine), not leaving it in gesture of the compiler